solutions/cf-1528A.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-1528A
+source: CF
+title: Parsa's Humongous Tree
+author: Ryan Chou
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/91058)
+
+## Explanation
+
+Let's go through a case where we've already computed the beauty values for all of vertex $u$'s neighbors.
+
+Let $\texttt{range}_u$ store $l_u$ and $r_u$ and $\texttt{adj}_u$ store the beauty values of all vertices adjacent to $u$.
+
+$\texttt{range}_u = \{1, 10\}$
+
+$\texttt{adj}_u = \{2, 3, 6, 9\}$
+
+Let's say that we select $7$, can we do better?
+
+The current beauty sum is: $|7-2| + |7-3| + |7-6| + |7-9| = 12$.
+
+Can we do better?
+
+Intuitively, for any $X$ that we choose to be $u$'s beauty value, if there's more elements that are greater than this $X$, making $X$ as small as possible would be more optimal (increasing differences for these elements) and vice versa. Note that in a case where the number of greater beauty values is the same as the number of smaller beauty values, setting $X$ to the minimum or maximum will never decrease the answer.
+
+It turns out, this observation is all we need. Since setting $u$ to $l_u$ or $r_u$ is always optimal, we'll handle this through DP.
```

Comment:
> Since setting $u$ to $l_u$ or $r_u$ is always optimal

elaborate?

Reply:
that was like the main point of the previous paragraph?
<img width="852" alt="image" src="https://user-images.githubusercontent.com/81596991/159806127-64fcbc61-0250-44ec-ae51-dc3e964a8e02.png">

Reply:
i... don't really get the last sentence lmao

Reply:
oops, ill make it clearer